### DISCUSSION TITLE
Support all Metadata Annotations for Json Schema Milestone

### DISCUSSION THREAD

**Discussion by jl-beast** [kubeapps/kubeapps#5512] [Ideas]
Currently from what I see in the Milestone, the implementation of the following basic metadata annotations from the [json-schema.org](http://json-schema.org/draft/2020-12/json-schema-validation.html) have not been mentioned:

- "deprecated"
  - could be implemented by showing a [deprecated] annotation to the far-right side of the renderer.
- "examples"
  - could be implemented for values by allowing for a drop-down underneath if examples are provided. When an Example is selected it would load it in. Would only be needed for string, number, and integer values.

the "readOnly" and "writeOnly" annotations would need to be specific to kubeApps, so I am not sure what purposes they would be used for.

**Comment by antgamdia** [comment]
Thanks for the feedback; definitely, seems like something useful and easy to implement.
Right now I'm working at https://github.com/vmware-tanzu/kubeapps/issues/3535, so that we can add/edit schemas more easily; this way, implementing these JSON Schema properties will be quicker.

**Reply by antgamdia** [reply]
I'm having a look at it now. For instance, using `readOnly` for disabling fields:

![image](https://user-images.githubusercontent.com/11535726/196945825-c15e1a9e-bc85-44a6-a356-ca10a0eb01ac.png)


Edit: and using the `examples`:

![image](https://user-images.githubusercontent.com/11535726/196978311-94b73102-2228-4b72-a874-22861aa70a81.png)
